packages/sdk/typescript/human-protocol-sdk/src/interfaces.ts

```diff
@@ -0,0 +1,29 @@
+import { BigNumber, Signer } from 'ethers';
+import { Provider } from '@ethersproject/abstract-provider';
+import { NetworkData } from './types';
+
+export interface IClientParams {
+  signerOrProvider: Signer | Provider;
+  network: NetworkData;
+}
+
+export interface IAllocation {
+  escrowAddress: string;
+  staker: string;
+  tokens: BigNumber;
+  createdAt: BigNumber;
+  closedAt: BigNumber;
+}
+
+export interface IReward {
+  escrowAddress: string;
+  slasher: string;
+  tokens: BigNumber;
+}
+
+export interface IStaker {
```

Comment:
is missing tokensAvailable

Reply:
This struct from packages/core/contracts/libs/Stakes.sol

struct Staker {
        uint256 tokensStaked; // Tokens staked by the Staker
        uint256 tokensAllocated; // Tokens allocated for jobs
        uint256 tokensLocked; // Tokens locked for withdrawal
        uint256 tokensLockedUntil; // Tokens locked until time
    }

Reply:
Unfortunately don't see the way how to get it, .tokensAvailable() is method of the Staker  structure inside the contract

Reply:
there is an internal function in stakes.sol, but you can calculate it yourself
tokensAvailable = tokensStaked - (tokensAllocated + tokensLocked)